README.adoc

```diff
@@ -10,3 +10,56 @@ image:http://www.javadoc.io/badge/{groupId}/{artifactId}.svg["Javadocs", link="h
 A Java project developed during the https://github.com/oliviercailloux/java-course[Java course] taught at Université Paris-Dauphine (in http://www.mido.dauphine.fr/[MIDO] M2 MIAGE).
 
 The code is hereby published under the MIT License, with their permission.
+
+
+== PLAN
+After the 4 iterations, we would like to improve the existing code as much as possible (optimization, new features).
+Two main categories of new features are to be considered:
+
+* a new interface system for voting preferences: set them up, implement them and create a new hierarchy of profiles.
+* a more complete file management system for preference representation: different formats, automatic exports and imports to manage longer and more complex preferences, possibility to use files present on the web.
+
+We will not focus on the GUI of the application.
+
+Of course, this is only a first approach. It will certainly evolve as the project progresses: we will of course keep you informed of any future modifications and clarifications.
+
+== Livraison 1
+The main purpose of this iteration is to appropriate the code already written, upgrade the project (change the groupid, switch to junit5, debug SWT for OSX and Windows, rename the packages, switch to pom oliviercailloux as a parent, etc.). Then clean the code as well as rewrite the equals(), hashCode() and tostring() functions and fix Eclipse warnings. Finally, we going to work on the 2 main improvements :
+
+* new preferences interfaces
+* import of profile files in ODS format respecting SOC format
+
+Tasks are not totally ordered in the description because they are not really dependent.
+
+=== Teams :
+* *SM/PO team* : Théo CHENNEBAULT
+* *Team 1* : Anael DRAY / Emilie CHHEAN
+* *Team 2* : Louis LEDOUX / Mattéo LOUBET
+
+=== Tasks
+.*Primary tasks* (not ordered)
+* 1#  README update for new groupeid and remove inappropriate things && modify LICENSE file
+* 2#  Pass to JUnit 5
+* 3#  Format files & code
+* 4#  Debug import SWT library for OSX and Windows
+* 5#  Rename all packages
+* 6#  Rework the Overide of toString(), equals() and hashCode()
+* 7#  Pass to parent pom oliviercailloux
+* 8#  Clean code and Eclipse Warnings suppression
+* 9#  Replace constructors with static methods
+* 10# Propose a series of interfaces for preferences. There are 8 interfaces in all : CompletePreference, AntiSymmetricCompletePreference, Preference, StrictPreference, MutablePreference, MutableStrictPreference, ImmutablePreference ImmutableStrictPreference
```

Comment:
I think that is too ambitious for a first iteration. And doing everything in one shot is associated with a much higher risk than doing it step by step: if you misunderstood the instructions, all your work could reveal not useful. Consider decomposing this task!

Also, consider submitting a proposal as a class diagram for example before implementing (a part of) this, to make sure that you and I have the same goal in mind.

Reply:
@oliviercailloux I don't know if this is too ambitious. I think we have understood your vision well, although (I prefer to tell you), the paragraph on "Preferences" interfaces is not very clear at first sight. Do you think that a partial rendering of a task is however feasible? By this I mean the rendering of the diagram and then the interfaces as they go along.

Reply:
Méfiez-vous. Les cimetières de l’ingénierie logicielle sont pleins de projets dont les développeurs avaient « parfaitement compris » la demande de leur client. (Bien sûr, j’admets qu’il est possible que sur ce coup, vous ayez effectivement bien compris. Mais on peut difficilement être sûr à l’avance.)

Vous pouvez très bien faire une tâche "diagrammes" et une tâche "interfaces", par exemple. Mais si vous souhaitez vraiment faire les deux ensemble et en faire une seule tâche, je vous laisse libre de faire ce choix. À vous de voir.